src/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -328,6 +328,17 @@ public T WithB2CAuthority(string authorityUri)
             return (T)this;
         }
 
+        /// <summary>
+        /// Sets the correlation id to be used in the authentication request.
+        /// </summary>
+        /// <param name="correlationId">correlation id of the authentication request</param>
+        /// <returns></returns>
+        public T WithCorrelationId(Guid correlationId)
```

Comment:
@jmprieur @henrik-me Using GUID here for the correlation ID. Do we want to keep it as is or change it to a string input instead? The authenticationResult in this PR also contains a correlation id of type GUID

Reply:
GUID is great

---
In reply to: [297839588](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1235#discussion_r297839588) [](ancestors = 297839588)

Reply:
MATS requires a guid for it's correlation ids, so my $.02 is that we make this a guid explicitly, as you have here.  That way we can format it appropriately for what the telemetry system and services are expecting (there's a AsMatsCorrelationId or something like that helper method in the .net mats telemetry code).